File: eval-reviews/2026-09-12-chained/workflow-recovery/typescript/fable/r1/starter/main.ts
import { readFileSync } from "node:fs";
import { LeasedSimulator, parseLeasedWorkflow } from "./leased.ts";
import { DomainError, parseWorkflow, Simulator } from "./workflow.ts";

function isLeased(input: unknown): boolean {
  return (
    input !== null &&
    typeof input === "object" &&
    !Array.isArray(input) &&
    Object.hasOwn(input, "workers")
  );
}
let response: unknown;
try {
  const request = JSON.parse(
    readFileSync(0, "utf8"),
    (_key: string, value: unknown, context?: { source?: string }) => {
      // Preserve the protocol's integer-token rule before JSON's numeric coercion loses it.
      if (
        typeof value === "number" &&
        context?.source &&
        !/^-?(0|[1-9][0-9]*)$/.test(context.source)
      )
        throw new DomainError("INVALID_INPUT");
      return value;
    },
  ) as { input?: unknown };
  const input = request?.input;
  const result = isLeased(input)
    ? new LeasedSimulator(parseLeasedWorkflow(input)).run()
    : new Simulator(parseWorkflow(input)).run();
  response = { ok: true, result };
} catch (error) {
  if (error instanceof DomainError)
    response = { ok: false, error: { code: error.code } };
  else if (error instanceof SyntaxError)
    response = { ok: false, error: { code: "INVALID_INPUT" } };
  else throw error;
}
process.stdout.write(JSON.stringify(response) + "\n");

File: eval-reviews/2026-09-12-chained/workflow-recovery/python/fable/r1/starter/main.py
"""One-request JSON adapter; each request owns a fresh simulator.

Requests with a `workers` field use the checkpoint-two leased-worker mode; all other
requests use the unchanged checkpoint-one contract.
"""
import json
import sys
from leased import LeasedSimulator, parse_leased_workflow
from workflow import DomainError, Simulator, parse_workflow


def main() -> None:
    try:
        request = json.load(sys.stdin)
        payload = request["input"]
        if type(payload) is dict and "workers" in payload:
            result = LeasedSimulator(parse_leased_workflow(payload)).run()
        else:
            result = Simulator(parse_workflow(payload)).run()
        response = {"ok": True, "result": result}
    except DomainError as error:
        response = {"ok": False, "error": {"code": error.code}}
    except (ValueError, KeyError, TypeError):
        response = {"ok": False, "error": {"code": "INVALID_INPUT"}}
    print(json.dumps(response, separators=(",", ":")))


if __name__ == "__main__":
    main()
